Python/Product/TestAdapter.Executor/UnitTest/TestDiscovererUnitTest.cs

```diff
@@ -105,9 +108,17 @@ internal class TestDiscovererUnitTest : IPythonTestDiscoverer {
             }
         }
 
-        private void CreateVsTests(IEnumerable<UnitTestDiscoveryResults> unitTestResults, IMessageLogger logger, ITestCaseDiscoverySink discoverySink) {
+        private void CreateVsTests(IEnumerable<UnitTestDiscoveryResults> unitTestResults,
+                                   IMessageLogger logger,
+                                   ITestCaseDiscoverySink discoverySink,
+                                   Dictionary<string, PythonProjectSettings> sourceToProjectSettings) {
             foreach (var test in unitTestResults?.SelectMany(result => result.Tests.Select(test => test)).MaybeEnumerate()) {
                 try {
+                    if(!sourceToProjectSettings.ContainsKey(test.Source)) {
+                        Warn(Strings.ErrorTestContainerNotFound.FormatUI(test.ToString()));
+                        continue;
+                    }
+
```

Comment:
unittest version is simpler because it's discovery doesn't modify the test.Source